addons/survey/models/survey_question.py

```diff
@@ -152,173 +152,108 @@ def _onchange_is_page(self):
-
-    def validate_datetime(self, post, answer_tag):
-        return self.date_validation('datetime', post, answer_tag, self.validation_min_datetime, self.validation_max_datetime)
-
-    def validate_simple_choice(self, post, answer_tag):
-        self.ensure_one()
-        errors = {}
-        if self.comments_allowed:
-            comment_tag = "%s_%s" % (answer_tag, 'comment')
-        # Empty answer to mandatory self
-        if self.constr_mandatory and answer_tag not in post:
-            errors.update({answer_tag: self.constr_error_msg})
-        if self.constr_mandatory and answer_tag in post and not post[answer_tag].strip():
-            errors.update({answer_tag: self.constr_error_msg})
-        # Answer is a comment and is empty
-        if self.constr_mandatory and answer_tag in post and post[answer_tag] == "-1" and self.comment_count_as_answer and comment_tag in post and not post[comment_tag].strip():
-            errors.update({answer_tag: self.constr_error_msg})
-        return errors
-
-    def validate_multiple_choice(self, post, answer_tag):
-        self.ensure_one()
-        errors = {}
-        if self.constr_mandatory:
-            answer_candidates = dict_keys_startswith(post, answer_tag)
-            comment_flag = answer_candidates.pop(("%s_%s" % (answer_tag, -1)), None)
-            if self.comments_allowed:
-                comment_answer = answer_candidates.pop(("%s_%s" % (answer_tag, 'comment')), '').strip()
-            # Preventing answers with blank value
-            if all(not answer.strip() for answer in answer_candidates.values()) and answer_candidates:
-                errors.update({answer_tag: self.constr_error_msg})
-            # There is no answer neither comments (if comments count as answer)
-            if not answer_candidates and self.comment_count_as_answer and (not comment_flag or not comment_answer):
-                errors.update({answer_tag: self.constr_error_msg})
-            # There is no answer at all
-            if not answer_candidates and not self.comment_count_as_answer:
-                errors.update({answer_tag: self.constr_error_msg})
-        return errors
-
-    def validate_matrix(self, post, answer_tag):
-        self.ensure_one()
-        errors = {}
+    def _validate_choice(self, answer, comment):
+        # Empty comment
         if self.constr_mandatory:
-            lines_number = len(self.labels_ids_2)
-            answer_candidates = dict_keys_startswith(post, answer_tag)
-            answer_candidates.pop(("%s_%s" % (answer_tag, 'comment')), '').strip()
-            # Number of lines that have been answered
-            if self.matrix_subtype == 'simple':
-                answer_number = len(answer_candidates)
-            elif self.matrix_subtype == 'multiple':
-                answer_number = len({sk.rsplit('_', 1)[0] for sk in answer_candidates})
+            if self.comments_allowed and self.comment_count_as_answer:
```

Comment:
Maybe?
```
if not answer and \
    not (self.comments_allowed and self.comment_count_as_answer and comment):
        return {self.id: self.constr_error_msg}
```

Reply:
- answer and comment: False
- answer and no comment: False
- no answer and comment + not configured: True
- no answer and comment + configured: False
- no answer and no comment + configured: True

That should cover it?